src/act.drive.cpp

```diff
@@ -1815,27 +1816,87 @@ void process_autonav(void)
         if (dir >= 0) {
           veh_moved = TRUE;
           move_vehicle(ch, dir);
+          
+          //Pull test for vehicles above * 1.5 load for stress damage happens on every room move and every 59 MUD minutes in save_vehicles().
+          // Rigger 3 p.64-65 & Shadowrun 3 p.145-147
+          if (veh->damage < VEH_DAM_THRESHOLD_DESTROYED && GET_VEH_ISOVERLOADED(veh) == LOAD_MAX) {
+            int tn = 8;
+            if (veh->damage) {
+              if (veh->damage >= VEH_DAM_THRESHOLD_SEVERE)
+                tn += 3;
+              else if (veh->damage >= VEH_DAM_THRESHOLD_MODERATE)
+                tn += 2;
+              else if (veh->damage >= VEH_DAM_THRESHOLD_LIGHT)
+                tn++;
+            }
+
+            if (success_test(veh->body, tn) < 1) {
+              veh->damage++;
+              send_to_char(ch, "%s got damaged by the stress of being immensely overloaded.\r\n", GET_VEH_NAME(veh));
+            }
+          }
+          if (veh->damage >= VEH_DAM_THRESHOLD_DESTROYED) {
+            veh_collapsed = TRUE;
+            break;
+          }
         }
       }
       if (!veh_moved) {
         send_to_veh("The autonav beeps and flashes a red 'ERROR' message before shutting off.\r\n", veh, 0, TRUE);
         veh->cspeed = SPEED_OFF;
         veh->dest = NULL;
-        return;
+        continue;
       }
       if (veh->in_room == veh->dest) {
         send_to_veh("Having reached its destination, the autonav shuts off.\r\n", veh, 0, TRUE);
         veh->cspeed = SPEED_OFF;
         veh->dest = NULL;
+        //Pull test for vehicles between * 1.0 and * 1.5 load for stress damage happens upon reaching destination and every 59 MUD minutes in save_vehicles().
+        // Rigger 3 p.64-65 & Shadowrun 3 p. 145-147
+        if (veh->damage < VEH_DAM_THRESHOLD_DESTROYED && GET_VEH_ISOVERLOADED(veh) == LOAD_HEAVY) {
+          int tn = 5;
+          if (veh->damage) {
+            if (veh->damage >= VEH_DAM_THRESHOLD_SEVERE)
+              tn += 3;
+            else if (veh->damage >= VEH_DAM_THRESHOLD_MODERATE)
+              tn += 2;
+            else if (veh->damage >= VEH_DAM_THRESHOLD_LIGHT)
+              tn++;
+          }
+            
+          if (success_test(veh->body, tn) < 1) {
+            veh->damage++;
+            send_to_char(ch, "%s got damaged by the stress of being immensely overloaded.\r\n", GET_VEH_NAME(veh));
+          }
+        }
+        if (veh->damage >= VEH_DAM_THRESHOLD_DESTROYED)
+          veh_collapsed = TRUE;
+      }
+    }
+    if (veh_collapsed) {
+      veh->cspeed = SPEED_OFF;
+      veh->dest = NULL;
+      struct char_data *tch = NULL, *next = NULL;
+      for (tch = veh->people; tch; tch = next) {
+        next = tch->next_in_veh;
+        char_from_room(tch);
+        char_to_room(tch, veh->in_room);
+        AFF_FLAGS(tch).RemoveBits(AFF_PILOT, AFF_RIG, ENDBIT);
+        send_to_char("Your vehicle collapses under load strain and you get out!\r\n", tch);
```

Comment:
Yeah I was thinking about that as well. But wanted to bounce it off you, Not sure if collapsing due to extreme weight is actually dangerous for the passengers or not. We're technically not crashing..it's more like the wheel axons (sorry not sure on English terminology on that stuff so not sure if I make sense heh) break and the chassis hits the ground, probably halting shortly with sparks flying off the asphalt... What do you think?

Reply:
At maximum speed, having your axles / shocks collapse will do a whole lot of damage and also make the car uncontrollable. It'd be worse than yanking the handbrake at highway speeds, because maximum speed here is far higher than highway speed, and the handbrake also doesn't drop the chassis into contact with the road and risk sending broken parts / fire into the cabin. This is a pretty traumatic experience for a car!

Reply:
Roger that. Will apply crash damage to passengers then like we do on the crash tests.